openhands/core/message.py

```diff
@@ -72,12 +73,7 @@ def serialize_model(self) -> dict:
         # - into a single string: for providers that don't support list of content items (e.g. no vision, no tool calls)
         # - into a list of content items: the new APIs of providers with vision/prompt caching/tool calls
         # NOTE: remove this when litellm or providers support the new API
-        if (
-            self.cache_enabled
-            or self.vision_enabled
-            or self.tool_call_id is not None
-            or self.tool_calls is not None
-        ):
+        if self.cache_enabled or self.vision_enabled or self.function_calling_enabled:
```

Comment:
```suggestion
        if (
            self.cache_enabled
            or self.vision_enabled
            or self.tool_call_id is not None
            or self.tool_calls is not None
        ):
```

Reply:
I think we need to revert these -- Since every message in CodeAct agent side is now being processed with pure function calling (i.e., every message is fn call format)